deeposlandia/network.py

```diff
@@ -5,20 +5,32 @@
 
 class ConvolutionalNeuralNetwork:
     """Convolutional neural network design
+
+    Attributes
+    ----------
+    _network_name : str
+        Name of the network
+    _image_size : integer
+        Input image size (height and width are equal)
+    _nb_channels : integer
+        Number of input image channels (1 for greyscaled images, 3 for RGB images)
+    _nb_labels : integer
+        Number of classes in the dataset glossary
+    X : tensor
+        (batch_size, image_size, image_size, nb_channels)-shaped input tensor; input image data
+
     """
 
     def __init__(self, network_name="mapillary", image_size=512,
                  nb_channels=3, nb_labels=65, learning_rate=1e-4):
-        """ Class constructor
-        """
         self._network_name = network_name
         self._image_size = image_size
         self._nb_channels = nb_channels
         self._nb_labels = nb_labels
-        self._X = K.layers.Input(shape=(image_size, image_size, nb_channels))
+        self.X = K.layers.Input(shape=(image_size, image_size, nb_channels), name="input")
 
     def convolution(self, x, nb_filters, kernel_size, strides=1, padding="same",
-                    activation='relu', batch_norm=True):
+                    activation='relu', batch_norm=True, name=""):
```

Comment:
The Keras layer, by default, defines a layer name according to its type, e.g. `Conv2D`, with an incremented integer such as `conv2d_2`, `conv2d_3`, and so on.

I think we should set it `None` by default. Then, Keras defines its own. And if we want to specify it, we can.

Reply:
Yes indeed, I saw the Keras default behavior for layer naming. `None` is a better idea than empty string, for sure. I change this.